Assets/Source/Components/Enemy/EnemyFormation.cs

```diff
@@ -0,0 +1,134 @@
+            {
+                List<int> row = new List<int>();
+                for(int y = 0; y < spawnLocationsY; y++)
+                {
+                    row.Add(0);
+                }
+                locations.Add(row);
+            }
+        }
+
+        public void UpdateLocations(List<List<int>> newLocations)
+        {
+            locations = newLocations;
+        }
+
+        public void SetLocation(int row, int column)
+        {
+            locations[row][column] = 1;
+        }
+
+        public void SetLocationRow(int row)
+        {
+            List<int> rowToUpdate = locations[row];
+            for(int i = 0; i < rowToUpdate.Count; i++)
+            {
+                rowToUpdate[i] = 1;
+            }
+        }
+
+        public void SetLocationColumn(int column)
+        {
+            foreach(List<int> row in locations)
+            {
+                row[column] = 1;
+            }
+        }
+
+        public void SpawnFormation()
+        {
+            //use formation and some magic to spawn, align, number, and group enemies together
+            //assume odd number
+            GameObject enemyPrafabToSpawn = GetRequiredResource<GameObject>($"{ResourcePaths.PrefabsFolder}/Actors/{GameObjects.Actors.ShooterEnemy}");
+            SpriteRenderer spriteRenderer = enemyPrafabToSpawn.GetComponent<SpriteRenderer>();
+            float width = spriteRenderer.size.x;
+            float height = spriteRenderer.size.y;
+
+            int rowIndex = 0;
+            foreach(List<int> row in locations)
+            {
+                int columnIndex = 0;
+                foreach(int i in row)
+                {
+                    if(i == 1)
+                    {
+                        float locationX = upperLeftLocation.x + columnIndex * width * 2;
+                        float locationY = upperLeftLocation.y - rowIndex * height * 2;
+                        Debug.Log("LocationX: " + locationX + "  upperLeft.x: " + upperLeftLocation.x + " index: " + columnIndex + " width: " + width);
+                        Debug.Log("LocationY: " + locationY + "  upperLeft.y: " + upperLeftLocation.y + " index: " + columnIndex + " height: " + width);
+
+                        GameObject enemy = InstantiatePrefab(enemyPrafabToSpawn, new Vector3(locationX, locationY, -10f));
+                        Debug.Log("Spawned Enemy at: " + enemy.transform.position);
+                    }
+                    columnIndex++;
+                }
+                rowIndex++;
+            }
+        }
+
+        public void PrintFormation()
+        {
+            StringBuilder sb = new StringBuilder();
+            foreach (List<int> row in locations)
+            {
+                
+                foreach(int column in row)
+                {
+                    sb.Append(column + " ");
+                }
+                sb.AppendLine();
+            }
+            Debug.Log(sb.ToString());
```

Comment:
i think we should delete these log statements for debugging after the code gets checked in, otherwise it just gets buried and never looked at again

Reply:
Yes sorry i must have missed this. Going through the whole solution and cleaning this up would be worth it. I saw you made a ticket for it.